sklearn/compose/_column_transformer.py

```diff
@@ -0,0 +1,591 @@
+"""
+The :mod:`sklearn.compose._column_transformer` module implements utilities
+to work with heterogeneous data and to apply different transformers to
+different columns.
+"""
+# Author: Andreas Mueller
+#         Joris Van den Bossche
+# License: BSD
+
+
+import numpy as np
+from scipy import sparse
+
+from ..base import clone, TransformerMixin
+from ..externals.joblib import Parallel, delayed
+from ..externals import six
+from ..pipeline import (
+    _fit_one_transformer, _fit_transform_one, _transform_one, _name_estimators)
+from ..preprocessing import FunctionTransformer
+from ..utils import Bunch
+from ..utils.metaestimators import _BaseComposition
+from ..utils.validation import check_is_fitted
+
+
+__all__ = ['ColumnTransformer', 'make_column_transformer']
+
+
+_ERR_MSG_1DCOLUMN = ("1D data passed to a transformer that expects 2D data. "
+                     "Try to specify the column selection as a list of one "
+                     "item instead of a scalar.")
+
+
+class ColumnTransformer(_BaseComposition, TransformerMixin):
+    """Applies transformers to columns of an array or pandas DataFrame.
+
+    EXPERIMENTAL: some behaviors may change between releases without
+    deprecation.
+
+    This estimator allows different columns or column subsets of the input
+    to be transformed separately and the results combined into a single
+    feature space.
+    This is useful for heterogeneous or columnar data, to combine several
+    feature extraction mechanisms or transformations into a single transformer.
+
+    Read more in the :ref:`User Guide <column_transformer>`.
+
+    .. versionadded:: 0.20
+
+    Parameters
+    ----------
+    transformers : list of tuples
+        List of (name, transformer, column) tuples specifying the transformer
+        objects to be applied to subsets of the data.
+
+        name : string
+            Like in FeatureUnion and Pipeline, this allows the transformer and
+            its parameters to be set using ``set_params`` and searched in grid
+            search.
+        transformer : estimator or {'passthrough', 'drop'}
+            Estimator must support `fit` and `transform`. Special-cased
+            strings 'drop' and 'passthrough' are accepted as well, to
+            indicate to drop the columns or to pass them through untransformed,
+            respectively.
+        column : string or int, array-like of string or int, slice or boolean \
+mask array
+            Indexes the data on its second axis. Integers are interpreted as
+            positional columns, while strings can reference DataFrame columns
+            by name.  A scalar string or int should be used where
+            ``transformer`` expects X to be a 1d array-like (vector),
+            otherwise a 2d array will be passed to the transformer.
+
+    unspecified : {'passthrough', 'drop'}, default 'drop'
+        By default, only the specified columns in `transformers` are
```

Comment:
single backticks? Is that referring to the glossary now? (Sorry I'm not caught up). Or do you mean double backticks?

Reply:
Yes, unless we enable a default_role, single backticks do nothing.

Reply:
ok. @jorisvandenbossche where did you get the idea about numpydoc? (@jnothman is probably the most active numpydoc dev so I kinda trust him on this ;)

Reply:
The last three review comments I made were "single backticks do nothing". Maybe we should enable that default role lol. But we could think about using glossary as default role instead of code as an alternative.

Reply:
single backticks make it look italic if you do not enable a default role (I think). 

In all example here https://numpydoc.readthedocs.io/en/latest/format.html#common-rest-concepts single backticks are used for parameter names. The section I linked to says this should also be done for variable, class, module names, but I suppose this is with the idea that the default role is to link to the API docs (which will never work for parameter names)